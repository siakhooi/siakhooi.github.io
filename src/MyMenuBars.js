import React from 'react'
import PropTypes from 'prop-types'

MyMenuBars.propTypes = {
  onClick: PropTypes.func
}

function MyMenuBars (props) {
  return (
    <button className="MyMenuBars fa fa-bars" onClick={() => props.onClick()} />
  )
}

export default MyMenuBars
