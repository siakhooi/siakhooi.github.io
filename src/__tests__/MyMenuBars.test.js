import React from 'react'
import renderer from 'react-test-renderer'
import MyMenuBars from '../MyMenuBars'

it('MyMenuBars/1', () => {
  const dummyFunction = jest.fn()

  const component =
    renderer.create(
      <MyMenuBars onClick={dummyFunction} />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  expect(dummyFunction).toHaveBeenCalled()
})
