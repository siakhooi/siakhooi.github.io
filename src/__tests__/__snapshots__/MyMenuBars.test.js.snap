// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMenuBars/1 1`] = `
<button
  className="MyMenuBars fa fa-bars"
  onClick={[Function]}
/>
`;
